packages/drivers/odsp-driver/src/opsCaching.ts

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// ISequencedDocumentMessage
+export interface IMessage {
+    sequenceNumber: number;
+}
+
+type CacheEntry = (IMessage | undefined)[];
+
+interface IBatch {
+    currentBatchSize: number;
+    batchData: CacheEntry;
+    dirty: boolean;
```

Comment:
Can you please also explain what does dirty mean with respect to batch. I am having a bit difficulty in understanding the logic of flushing ops w.r.t to dirty.
This would help me in understanding the logic.